sotodlib/site_pipeline/make_filterbin_map.py

```diff
@@ -0,0 +1,1144 @@
+                                            inclusive=False, op=np.var),
+                         10, -1)**0.5
+    typical = np.median(rms)
+    # Take away dets with extreme noise
+    rmstol = 10
+    good = (rms>typical/rmstol)&(rms<typical*rmstol)
+    obs.restrict("dets", obs.dets.vals[good])
+    rms = rms[good]
+    # Build cuts, based on maximum allowed fraction of data values
+    # above the threshold.
+    cuttol = 1000
+    maxcut = 0.05
+    cutmask = np.abs(obs.signal)>rms[:,None]*cuttol
+    cutfrac = np.mean(cutmask,1)
+    good = cutfrac <= maxcut
+    print("all dets: ", obs.dets.count)
+    print("good: ", good.shape)
+    obs.restrict("dets", obs.dets.vals[good])
+
+
+def wrap_info(obs, site):
+    obs.wrap("weather", np.full(1, "toco"))
+    obs.wrap("site",    np.full(1, site))
+
+    # Union of flags into glitch_flags.
+    # Use this if not using pre-process database:
+    # (glitch_flags full of 0s when we don't have the preprocess data base)
+    obs.flags.wrap('glitch_flags', so3g.proj.RangesMatrix.zeros(obs.shape[:2]),
+                   [(0, 'dets'), (1, 'samps')])
+    # else:
+    #obs.flags.wrap('glitch_flags', obs.preprocess.turnaround_flags.turnarounds
+    #               + obs.preprocess.jumps_2pi.jump_flag + obs.preprocess.glitches.glitch_flags, )
+    ## TODO add --preprocess to args
+    return
+
+
+def get_detector_cuts_flags(obs, rfrac_range=(0.05,0.9), psat_range=(0,20)):
+    try:
+        flags.get_turnaround_flags(obs) 
+    except Exception as e:
+        return False
+    flags.get_det_bias_flags(obs, rfrac_range=rfrac_range, psat_range=psat_range)
+    bad_dets = has_all_cut(obs.flags.det_bias_flags)
+    obs.restrict('dets', obs.dets.vals[~bad_dets])
+    #if obs.dets.count<=1: return obs # check if I cut all the detectors after the det bias flags
+    return True
+            
+def select_data(obs,
+                det_left_right, det_in_out, det_upper_lower):
+
+    # Restrict non optical detectors, which have nans in their focal plane
+    # coordinates and will crash the mapmaking operation.
+    obs.restrict('dets', obs.dets.vals[obs.det_info.wafer.type == 'OPTC'])
+
+    status = get_detector_cuts_flags(obs)
+    if not(status):
+        return False
+
+    #if obs.dets.count<=1: return obs # check if I cut all the detectors after the det bias flags
+
+    return True
+
+
+def subtract_hwpss(obs):
+    hwp.get_hwpss(obs)
+    hwp.subtract_hwpss(obs)
+
+    
+def get_trending_cuts(obs):
+    # Note: n_pieces should be dependent on the length of the observation
+    flags.get_trending_flags(obs,
+                             n_pieces=10,
+                             max_trend=2.5) #,max_samples=obs.samps.count)
+    tdets = has_any_cuts(obs.flags.trends)
+    obs.restrict('dets', obs.dets.vals[~tdets])
+    #print('dets after trending cuts: ', {obs.dets.count})
+
+    
+def get_jumps(obs, signal_name):
+    jflags, _, jfix = jumps.twopi_jumps(obs,
```

Comment:
This function is failing for some cases, I'll send you the script with the correction.